### DISCUSSION TITLE
Is it possible to automatically load `plugin/*.fnl` files?

### DISCUSSION THREAD

**Discussion by iagoleal** [rktjmp/hotpot.nvim#119] [Q&A] [answered]
Hello!
It has been quite some time since #44 and with hotpot 0.9 there is now support for automatically loading `ftplugin/*.fnl`. Thus, it got me wondering if is there any way to also automatically load `plugin/*.fnl` files.

I tried putting `runtime! plugin/**/*.fnl` in my config but got an error message 

    could not find module path for require command (not descendent of a 'fnl' dir?)
    
Is there any way to circumvent this 'fnl' dir limitation? From what I understood, [nvim-moonwalk](https://github.com/gpanders/nvim-moonwalk) is inspired by hotpot and supports this. So I imagine there must be some way to also do that from hotpot.

Thanks!

**Comment by rktjmp** [comment]
Honestly its not a nvim feature I use a lot. I think we can hook the `SourecCmd` autocommand and do what we do for FTPlugins, but I am not sure if that will count as "automatic". Eg, does nvim automatically run that, and can we hook it as we want, or does it depend on you putting a call to `runtime!` in your config -- which is still sort of automatic, or more automatic than right now!

I think it should also include `after/` dirs? Not 100% on how nested those can be (eg: `after/ftplugin/x.fnl`). They might be tricky.

Its also not without some cost since we need to search all the dirs for files that match `plugins/**/*.fnl`, but I also yolo'd that consideration away for ftplugins -- I think the default-on behaviour is much nicer and probably better to expose an "opt out" switch if needed than requiring opt in.

If your ok with ~~filthy disgusting~~ lua files in your dirs, you can (should be able to?) get it working now with a `.hotpot.lua` file or similar autocommand as described in the "init.fnl" section of the cookbook.

ref: https://github.com/rktjmp/hotpot.nvim/issues/121

**Reply by rktjmp** [reply, marked as answer]
I believe it should be working as of fc8715eca5a81072e9a52f7c3afc8574d73b4d9c.

There are some quirks, we can only run after all of neovims loading has finished, so `plugin/x.fnl` is actually run *after* `after/plugin/x.lua`. There isn't any way around that but if its an issue you can compile to `plugin/x.lua` with `.hotpot.lua` or `api.make`. I think for most people this wont matter...

**Reply by iagoleal** [reply]
Thanks, it works like a charm!
I like using the `plugin/` folder for modularizing parts of my config instead of calling a lot of `require`s. Now I don't some lua wrappers for that anymore!

**Reply by rktjmp** [reply]
Great! As I said, I dont *really* use that feature much -- probably because hotpot didn't support it! If you hit bugs or oddities definitely open an issue.

**Comment by rktjmp** [comment]
Discussion not relevant to v2. Support is built in.
